packages/fiori/src/UserMenu.ts

```diff
@@ -0,0 +1,335 @@
+	 * When using this attribute in a declarative way, you must only use the `id` (as a string) of the element at which you want to show the popover.
+	 * You can only set the `opener` attribute to a DOM Reference when using JavaScript.
+	 * @public
+	 * @default undefined
+	 */
+	@property({ converter: DOMReferenceConverter })
+	opener?: HTMLElement | string;
+
+	/**
+	 * Defines if the User Menu shows the Manage Account option.
+	 *
+	 * @default false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	showManageAccount = false;
+
+	/**
+	 * Defines if the User Menu shows the Other Accounts option.
+	 *
+	 * @default false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	showOtherAccounts = false;
+
+	/**
+	 * Defines the menu items.
+	 * @public
+	 */
+	@slot({
+		type: HTMLElement,
+		"default": true,
+	})
+	menuItems!: Array<UserMenuItem>;
+
+	/**
+	 * Defines the user accounts.
+	 *
+	 * **Note:** If one item is used, it will be shown as the selected one. If more than one item is used, the first one will be shown as selected unless
+	 * there is an item with `selected` property set to `true`.
+	 * @public
+	 */
+	@slot({
+		type: HTMLElement,
+		invalidateOnChildChange: {
+			properties: true,
+			slots: false,
+		},
+	})
+	accounts!: Array<UserMenuAccount>;
+
+	@i18n("@ui5/webcomponents-fiori")
+	static i18nBundle: I18nBundle;
+
+	/**
+	 * @private
+	 */
+	_selectedAccount!: UserMenuAccount;
+
+	/**
+	 * @private
+	 */
+	_otherAccounts: UserMenuAccount[] = [];
+
+	onBeforeRendering() {
+		this._selectedAccount = this.accounts.find(account => account.selected) || this.accounts[0];
+		this._otherAccounts = this.accounts.filter(account => account !== this._selectedAccount);
+	}
+
+	_handleAvatarClick() {
+		this.fireDecoratorEvent("avatar-click");
+	}
+
+	_handleManageAccountClick() {
+		this.fireDecoratorEvent("manage-account-click");
+	}
+
+	_handleAddAccountClick() {
+		this.fireDecoratorEvent("add-account-click");
+	}
+
+	_handleAccountSwitch(e: CustomEvent<{ item: ListItemClickEventDetail & { associatedAccount: UserMenuAccount } }>) {
+		const eventPrevented = !this.fireDecoratorEvent<UserMenuOtherAccountClickEventDetail>("change-account", {
+			prevSelectedAccount: this._selectedAccount,
+			selectedAccount: e.detail.item.associatedAccount,
+		});
+
+		if (eventPrevented) {
+			return;
+		}
+
+		this._selectedAccount.selected = false;
+		e.detail.item.associatedAccount.selected = true;
+	}
+
+	_handleSignOutClick() {
+		const eventPrevented = !this.fireDecoratorEvent("sign-out-click");
+
+		if (eventPrevented) {
+			return;
+		}
+
+		 this._closeUserMenu();
+	}
+
+	_handleMenuItemClick(e: CustomEvent<UserMenuItemClickEventDetail>) {
+		const item = e.detail.item;
+
+		if (!item._popover) {
+			const eventPrevented = !this.fireDecoratorEvent<UserMenuItemClickEventDetail>("item-click", {
+				"item": item,
+			});
+
+			if (!eventPrevented) {
+				item.fireEvent("close-menu");
```

Comment:
Custom events have description about their bubble, cancelable configurations. This means that we need to fire events via `fireDecoratorEvent` in order to apply these configurations

Reply:
For some reason, it is not working. I guess because the MenuItem (which we extend) is still using the fireEvent. We can sync for that

Reply:
Most probably this will be resolved when the following PR is merged https://github.com/SAP/ui5-webcomponents/pull/10247/files

Reply:
Yes, Adi, you are right - @tsanislavgatev